assertj-core/src/main/java/org/assertj/core/api/recursive/comparison/FieldLocation.java

```diff
@@ -163,7 +166,10 @@ public boolean equals(Object obj) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(pathToUseInRules, decomposedPath, pathsHierarchyToUseInRules);
+    int result = Objects.hashCode(pathToUseInRules);
+    result = 31 * result + Objects.hashCode(decomposedPath);
+    result = 31 * result + Objects.hashCode(pathsHierarchyToUseInRules);
+    return result;
```

Comment:
@schlosna is there a reason not to only use `pathToUseInRules` as I suggested ? Was there any issue doing that ?

Reply:
No biggie as this implementation is fine anyway, and the performance looks good on my machine.

Reply:
The [equals verifier test was failing](https://github.com/schlosna/assertj-core/actions/runs/7671553403/job/20910160863#step:4:6666) as `hashCode()` did not use all of the fields that `equals` does, so I rolled `hashCode` back per https://github.com/ash211/assertj/pull/1#discussion_r1468652427 .

If we really want, we could do something like following so that the other fields contribute to hash code in a cheap way, but I'm inclined to leave it as is unless `Objects.hashCode` becomes a further bottleneck.


```
  @Override
  public int hashCode() {
    int result = Objects.hashCode(pathToUseInRules);
    // The other fields are derivations of pathToUseInRules, so not truly necessary for hashCode.
    // To make EqualsVerifier happy, we can use collection sizes as cheap mechanism to contribute to hash code.
    result = 31 * result + (decomposedPath == null ? 0 : decomposedPath.size());
    result = 31 * result + (pathsHierarchyToUseInRules == null ? 0 : pathsHierarchyToUseInRules.size());
    return result;
  }
```

Reply:
Fine by me, I tried using only `pathToUseInRules` but it did not make a noticeable improvement performance wise.